basket/main.py

```diff
@@ -108,7 +147,7 @@ def _get_requirements_from_tar_archive(self, path):
         TAR-gzipped or TAR-bzipped archive.
         """
         extension = self.os.path.splitext(path)[1].split('.')[1]
-        with self.tarfile.open(path, 'r:%s' % extension) as archive:
+        with closing(self.tarfile.open(path, 'r:%s' % extension)) as archive:
```

Comment:
Oh, scratch that. I missed the message of your commit. These context managers have been added in Python 2.7. This change adds support for Python 2.6. Good catch! :)